yarn-project/cli/src/cmds/misc/setup_contracts.ts

```diff
@@ -17,9 +17,22 @@ export async function setupCanonicalL2FeeJuice(
   const { FeeJuiceContract } = await import('@aztec/noir-contracts.js');
 
   const feeJuiceContract = await FeeJuiceContract.at(ProtocolContractAddress.FeeJuice, deployer);
+
   log('setupCanonicalL2FeeJuice: Calling initialize on fee juice contract...');
-  await feeJuiceContract.methods
-    .initialize(feeJuicePortalAddress)
-    .send({ fee: { paymentMethod: new NoFeePaymentMethod(), gasSettings: GasSettings.teardownless() } })
-    .wait(waitOpts);
+
+  try {
+    const provenTx = await feeJuiceContract.methods
+      .initialize(feeJuicePortalAddress)
+      .prove({ fee: { paymentMethod: new NoFeePaymentMethod(), gasSettings: GasSettings.teardownless() } });
+
+    await provenTx.send().wait(waitOpts);
+    log('setupCanonicalL2FeeJuice: Fee juice contract initialized');
+  } catch (e: any) {
+    if (e instanceof Error && e.message.includes('(method pxe_simulateTx) (code 400) Assertion failed')) {
```

Comment:
Noting that this looks almost guaranteed to churn at some point (method name change, format change etc)

Reply:
correct solution would be maybe 204 error code for this case

Reply:
Added a todo for now